networkx/algorithms/approximation/steinertree.py

```diff
@@ -0,0 +1,75 @@
+from itertools import combinations, chain
+
+from networkx.utils import pairwise, not_implemented_for
+import networkx as nx
+
+__all__ = ['metric_closure', 'steiner_tree']
+
+
+@not_implemented_for('directed')
+def metric_closure(G, weight='weight'):
+    """  Return the metric closure of a graph.
+
+    The metric closure of a graph G is the complete graph in which each edge is
+    weighted by the shortest path distance between the nodes in G.
+
+    Parameters
+    ----------
+    G : Networkx graph
+
+    Returns
+    -------
+    Networkx graph
+        Metric closure of the graph G.
+
+    """
+    M = nx.Graph()
+
+    for u, v in combinations(G, 2):
+        distance, path = nx.single_source_dijkstra(G, u, v, weight=weight)
+        M.add_edge(u, v, distance=distance[v], path=path[v])
+
+    return M
+
+
+@not_implemented_for('directed')
+def steiner_tree(G, terminal_nodes, weight='weight'):
+    """ Returns an approximation to minimum steiner tree of the graph G.
```

Comment:
[PEP 257][1] suggests using the imperative mood for the summary phrase.

> The docstring is a phrase ending in a period. It prescribes the function or method's effect as a command ("Do this", "Return that"), not as a description; e.g. don't write "Returns the pathname ...".

So this should be `Return an approximation to the minimum Steiner tree of a graph`.

Steiner, a name, is a proper noun so it should capitalized.

[1]: https://www.python.org/dev/peps/pep-0257/